src/quantuminspire/qiskit/backend_qx.py

```diff
@@ -282,36 +284,76 @@ def _collect_measurements(experiment):
             measurements = [[index, index] for index in range(number_of_qubits)]
         return {'measurements': measurements, 'number_of_clbits': number_of_clbits}
 
+    @staticmethod
+    def __qubit_2_classical_hex(qubit_register, measurements, number_of_qubits):
+        """ this function converts the qubit register data to the hexadecimal representation of the classical state
+
+        Args:
+            qubit_register (int): The measured value of the qubits represented as int
+            measurements (dict): Measured qubits/classical bits map and number of classical bits
+            number_of_qubits (int): number of qubits used in the algorithm
+
+        Returns:
+            str: The hexadecimal value of the classical state.
+
+        """
+        qubit_state = ('{0:0{1}b}'.format(int(qubit_register), number_of_qubits))
+        classical_state = ['0'] * measurements['number_of_clbits']
+        for q, c in measurements['measurements']:
+            classical_state[c] = qubit_state[q]
+        classical_state = ''.join(classical_state)
+        classical_state_hex = hex(int(classical_state, 2))
+        return classical_state_hex
+
     @staticmethod
     def __convert_histogram(result, measurements, number_of_qubits, number_of_shots):
         """ The quantum inspire backend always uses full state projection. The SDK user
             can measure not all qubits and change the combined classical bits. This function
             converts the histogram output, such that it represents the counts measured with the
             classical bits.
 
-            Args:
-                result (dict): The result output from the quantum inspire backend with full-
-                               state projection histogram output.
-                measurements (dict): Measured qubits/classical bits map and number of classical bits
-                number_of_qubits (int): number of qubits used in the algorithm
-                number_of_shots (int): The number of times the algorithm is executed.
+        Args:
+            result (dict): The result output from the quantum inspire backend with full-
+                           state projection histogram output.
+            measurements (dict): Measured qubits/classical bits map and number of classical bits
```

Comment:
```suggestion
            measurements (list): Measured qubits/classical bits map and number of classical bits
```

Reply:
Actually what was meant here is about the field 'measurements' in the dictionary with the same name 'measurements', the field measurements must be a list (and it is).